theme.js

```diff
@@ -0,0 +1,40 @@
+import { slug } from 'github-slugger'
+import path from 'node:path'
+import {
+  MarkdownTheme,
+  MarkdownThemeRenderContext
+} from 'typedoc-plugin-markdown'
+
+const externalLinkRegex = /^(http|ftp)s?:\/\//
+
+export function load(app) {
+  app.renderer.defineTheme('custom-markdown-theme', CustomMarkdownTheme)
+}
+
+export class CustomMarkdownTheme extends MarkdownTheme {
+  getRenderContext(pageEvent) {
+    return new CustomMarkdownThemeContext(pageEvent, this.application.options)
+  }
+}
+
+class CustomMarkdownThemeContext extends MarkdownThemeRenderContext {
+  relativeURL = url => {
+    if (!url) {
+      return null
+    } else if (externalLinkRegex.test(url)) {
+      return url
+    }
+
+    let filePath = path.parse(url)
+    let [, anchor] = filePath.base.split('#')
+    let basePath = this.options.getValue('basePath')
+
+    let constructedUrl = typeof basePath === 'string' ? basePath : ''
```

Comment:
Will it work for other links, not from TypeDoc?

Reply:
Could you please explain your question?

Reply:
This code is TypeDoc-specific and will be applied to TypeDoc-generated documents.

Reply:
Great. In this case it is not a problem.